l10n_es_dua_sii/models/account_invoice.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2017 Consultoría Informática Studio 73 S.L.
+# Copyright 2017 Comunitea Servicios Tecnológicos S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api, exceptions, _
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.model
+    def _get_dua_fiscal_position(self):
+        company = self.env.user.company_id
+        dua_fiscal_position = self.env.ref(
+            'l10n_es_dua.%i_fp_dua' % (company.id),
+            raise_if_not_found=False
+        ) or self.env['account.fiscal.position'].search([
+            ('name', '=', 'Importación con DUA')
+        ])
+        if not dua_fiscal_position:
+            raise exceptions.UserError(
+                _("To use DUA and SII you need fiscal position named "
+                  "'Importación con DUA'."))
+        return dua_fiscal_position
+
+    @api.multi
+    @api.depends('tax_line_ids')
+    def _compute_sii_enabled(self):
+        dua_fiscal_position = self._get_dua_fiscal_position()
+        for invoice in self:
+            if invoice.fiscal_position_id == dua_fiscal_position and \
+                    not invoice.sii_dua_invoice:
+                invoice.sii_enabled = False
+            else:
+                super(AccountInvoice, invoice)._compute_sii_enabled()
+
+    @api.multi
+    def _compute_dua_invoice(self):
+        dua_fiscal_position = self._get_dua_fiscal_position()
+        for invoice in self:
+            if invoice.fiscal_position_id == dua_fiscal_position and \
+                    invoice.tax_line_ids.\
+                    filtered(lambda x: x.tax_id.description in
```

Comment:
Sería mejor crear una línea de mapeo para esto.

Reply:
También se puede hacer así:
`invoice.tax_line_ids.filtered(lambda x: x.tax_id.description not in
                                             ['DUA Exento']):
`